src/tools/tests/plan-generator-tests.ts

```diff
@@ -153,5 +166,26 @@ describe('recipe2plan', () => {
         actuals.push(newActual);
       }
     });
+    it('uses the same identifier for all HandleConnections connected to the same handle', async () => {
+      const manifest = await Manifest.parse(`\
+     particle A
+       data: writes Thing {num: Number}
+     particle B
+       data: reads Thing {num: Number}
+       
+     recipe R
+       h: create persistent 
+       A
+         data: writes h
+       B
+         data: reads h`);
+      const recipeResolver = new StorageKeyRecipeResolver(manifest);
+      const recipes = await recipeResolver.resolve();
+      const generator = new PlanGenerator(recipes, '');
+      const plan = await generator.generate();
+      const keys = collectOccurrences(plan, 'CreateableStorageKey("', '")');
```

Comment:
This looks fragile. I presume it would be possible to refactor the code generated Kotlin in such a way that `CreateableStorageKey` was present only once and referred to from both handle connections, and then it would still be correct but the test would fail, or am I wrong?

Reply:
If there is no better way you can imagine testing it it's ok. What would be good is adding 2 readers to a test with golden file though.

Reply:
I can add two readers to the golden file, but I want to avoid depending on randomness for those tests.

Reply:
> I presume it would be possible to refactor the code generated Kotlin in such a way that CreateableStorageKey was present only once and referred to from both handle connections, and then it would still be correct but the test would fail, or am I wrong?

I could do this. The `CreateableStorageKey` is a dataclass, so equality between two instances is derived from the key. As long as those are the same, they are considered equal. So, the means that I have now is sufficient. I'm leaning to keeping it the same because I like the pattern of composed instantiation instead of referring to private properties on the object. Plus, the generator right now depends on the hierarchy of components in the manifest; it would be tricky to extract out a function to generate the keys in a vacuum.

Reply:
It used to be that way until my original graph data structure got refactored. I originally had a top level set of vertices (Particles, Handles) in the plan and a set of Edges (HandleConnection), but now the 'Handles' field got eliminated. 

CreateableStorageKey has strong equality semantics. 

But if you wanted to also do same-instance, you could do 

val createableKey = CreatealeStorageKey('blah')
and then pass createableKey to the HandleConnection constructors. I won't matter to Allocator, since these are immutable datastructures.